src/main/java/domain/BlackjackRule.java

```diff
@@ -0,0 +1,43 @@
+package domain;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class BlackjackRule {
+    private static final int BUST_CONDITION = 21;
+
+    private final Map<Player, Entry<Integer, Integer>> results = new LinkedHashMap<>();
+
+    public BlackjackResult finishGame(final List<Player> players, final Player dealer) {
+        results.put(dealer, Map.entry(0, 0));
+        players.forEach(player -> {
+            results.put(player, Map.entry(0, 0));
+            judgeWinningHand(dealer, player);
+        });
+        return new BlackjackResult(results);
+    }
+
+    private void judgeWinningHand(final Player dealer, final Player player) {
+        if (player.calculateScore() > BUST_CONDITION) {
```

Comment:
player에게 bust 상태인지 물어보는 메시지를 보내면 어떨까요!
```suggestion
        if (player.isBust()) {
```